### PR TITLE
WIP: add chain tdnn AMI recipe

### PR DESCRIPTION
**Pull request by freewym**
@vijayaditya

### CODE REVIEW

File: egs/ami/s5/local/chain/run_tdnn_py.sh [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+dir=exp/chain/$mic/tdnn${affix:+_$affix} # Note: _sp will get added to this if $speed_perturb == true.
+
+# training options
+num_epochs=4
+initial_effective_lrate=0.001
+final_effective_lrate=0.0001
+leftmost_questions_truncate=-1
+max_param_change=2.0
+final_layer_normalize_target=0.5
+num_jobs_initial=3
```
Comment (vijayaditya) [inline egs/ami/s5/local/chain/run_tdnn_py.sh]: num_jobs_initial=2
num_jobs_final=12

File: egs/ami/s5/local/chain/run_tdnn_py.sh [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+num_jobs_final=16
+minibatch_size=128
+frames_per_eg=150
+remove_egs=false
```
Comment (vijayaditya) [inline egs/ami/s5/local/chain/run_tdnn_py.sh]: remove_egs=true

File: egs/ami/s5/local/chain/run_tdnn_py.sh [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 12 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate $leftmost_questions_truncate \
+      --cmd "$train_cmd" 9000 data/$mic/$train_set $lang $ali_dir $treedir
```
Comment (vijayaditya) [inline egs/ami/s5/local/chain/run_tdnn_py.sh]: You probably don't need 9000 leaves. You could start with 6000, may be.
@danpovey Could you suggest a value for 80 hrs of training data.

File: egs/ami/s5/local/chain/run_tdnn_py.sh [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+fi
+
+if [ $stage -le 13 ]; then
+  echo "$0: creating neural net configs";
```
Comment (vijayaditya) [inline egs/ami/s5/local/chain/run_tdnn_py.sh]: Could you use the same TDNN config as the normal local/nnet3/run_tdnn.sh script.

Comment (danpovey) [inline egs/ami/s5/local/chain/run_tdnn_py.sh]: I agree that 9k is too much.  I suggest to take whatever value was used in
the regular  nnet2 or nnet3 recipes and decrease it by about 10%.

**Comment by vijayaditya** [comment]
@danpovey  good to merge.. there are minor modifications necessary to simplify the script but it is easier for me to do it .. once the merge is done..

**Pull request merged**